mat_dp_pipeline/data_sources/tmba.py

```diff
@@ -38,52 +27,30 @@ def __init__(
         target_csv: Path,
         targets_parameters: list[str],
         country_source: type[SourceWithCountries],
-        parameter_to_category: dict[str, str] | None = None,
-        variable_to_specific: dict[str, str | None] | None = None,
     ):
         self._targets_csv = target_csv
         self._targets_parameters = targets_parameters
-        self._parameter_to_category = (
-            parameter_to_category if parameter_to_category else PARAMETER_TO_CATEGORY
-        )
-        self._variable_to_specific = (
-            variable_to_specific if variable_to_specific else VARIABLE_TO_SPECIFIC
-        )
         self._country_to_path = country_source.country_to_path(
             identifier=Identifier.alpha_2
         )
 
     def __call__(self, output_dir: Path) -> None:
         targets = pd.read_csv(self._targets_csv)
         targets = targets[targets["parameter"].isin(self._targets_parameters)]
-        targets = (
-            targets.drop(columns=[targets.columns[0], "scenario"])
-            .rename(columns={"variable": "Specific"})
-            .dropna()
-        )
-        category = targets["parameter"].map(self._parameter_to_category)
-        targets.insert(0, "Category", category)
-        for pattern, replacement in self._variable_to_specific.items():
-            # Remove (ignore) if None replacement, else replace
-            if replacement is None:
-                targets = targets[targets["Specific"] != pattern]
-            else:
-                targets["Specific"] = targets["Specific"].str.replace(
-                    pattern, replacement
-                )
+        targets = targets.drop(columns=[targets.columns[0], "scenario"]).dropna()
+        targets = map_technologies(targets, "variable", self._tech_map)
 
-        # TODO: move this. to __init__?
-        grouping = ["country", "parameter"]
+        grouping = list(self._grouping)  # list needed
```

Comment:
This tuple is only used in one place I think (here) - appreciate the whole immutable values thing but the conversion seems unnecessary when we could have started with a list

Reply:
Fair rnough